### PR TITLE
Mf 354 create legal entity tests

### PR DESCRIPTION
**Pull request by robinoconnor**

### CODE REVIEW

File: src/SFA.DAS.Reservations.Functions.LegalEntities.AcceptanceTests/TestServiceProvider.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            var serviceCollection = new ServiceCollection();
+            var configuration = GenerateConfiguration();
+
+            var serviceProviderBuilder = new ServiceProviderBuilder(new LoggerFactory(), configuration)
+            {
+                ServiceCollection = serviceCollection
+            };
+            
+            var encodingService = new Mock<IEncodingService>();
+            encodingService.Setup(x => x.Decode(It.Is<string>(s => s.Equals(TestDataValues.NonLevyHashedAccountId)),It.IsAny<EncodingType>())).Returns(TestDataValues.NonLevyAccountId);
```
Comment (dashton82) [inline src/SFA.DAS.Reservations.Functions.LegalEntities.AcceptanceTests/TestServiceProvider.cs]: With these setups i'd make sure that its using the correct EncodingType and not an `It.IsAny<>`

File: src/SFA.DAS.Reservations.Functions.LegalEntities.AcceptanceTests/Features/LegalEntityUpdated.feature [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Scenario: A existing legal entity has recieved levy
+	Given I have an existing unsigned, non levy legal entity
+	When levy added event is triggered
+	Then the legal entity should be marked as a levy entity
+
+Scenario: A new legal entity has recieved levy
+	Given I have a legal entity that is new
+	When levy added event is triggered
+	Then the legal entity should not be available
```
Comment (dashton82) [inline src/SFA.DAS.Reservations.Functions.LegalEntities.AcceptanceTests/Features/LegalEntityUpdated.feature]: Not sure these last two Thens are really describing the business functionality

File: src/SFA.DAS.Reservations.Application/AccountLegalEntities/Handlers/LevyAddedToAccountHandler.cs [hunk 3]
```diff
@@ -7,15 +9,25 @@ namespace SFA.DAS.Reservations.Application.AccountLegalEntities.Handlers
...
         public async Task Handle(LevyAddedToAccount levyAddedToAccountEvent)
         {
-            await _accountLegalEntitiesService.UpdateAccountLegalEntitiesToLevy(levyAddedToAccountEvent);
+            try
+            {
+                await _accountLegalEntitiesService.UpdateAccountLegalEntitiesToLevy(levyAddedToAccountEvent);
+            }
+            catch (DbUpdateException e)
+            {
+                _logger.LogWarning("Could not update agreement status to signed", e);
```
[the remark this answers is not in the document]
Comment (robinoconnor) [reply src/SFA.DAS.Reservations.Application/AccountLegalEntities/Handlers/LevyAddedToAccountHandler.cs]: ok, i'll change these then based on that new approach as i know originally we didn't want any retries (at least on the other previous message handlers)

File: src/SFA.DAS.Reservations.Functions.LegalEntities.AcceptanceTests/Features/LegalEntityUpdated.feature (hunk 2 above)
Comment (robinoconnor) [reply src/SFA.DAS.Reservations.Functions.LegalEntities.AcceptanceTests/Features/LegalEntityUpdated.feature]: I couldn't come up with a good description that matched both what was happening and what the business output was but i'll have a rethink and see if I can come up with something better

**Pull request merged**